engine/runtime-compiler/src/test/java/org/enso/compiler/test/mock/VirtualFileSystem.java

```diff
@@ -0,0 +1,218 @@
+package org.enso.compiler.test.mock;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.nio.file.attribute.FileTime;
+import java.util.Arrays;
+import java.util.stream.Stream;
+import org.apache.commons.vfs2.AllFileSelector;
+import org.apache.commons.vfs2.FileObject;
+import org.apache.commons.vfs2.FileSystemException;
+import org.apache.commons.vfs2.FileSystemManager;
+import org.apache.commons.vfs2.VFS;
+import org.enso.filesystem.FileSystem;
+
+/**
+ * {@link FileSystem} implementation for Apache Commons VFS2. So far, only RAM file system is
+ * supported.
+ */
+final class VirtualFileSystem implements FileSystem<FileObject> {
```

Comment:
- using 3rd party API class in the signature is diverging us from my dreamed of direction
- using `java.nio.Path` instead in the API signature would avoid such 3rd party API dependency
- using [jimfs](https://github.com/google/jimfs) as the in memory implementation would be 1:1 rewrite

Reply:
Replaced Apache Commons VFS2 with com.google.jimfs in ae8e0f985b03dcdafc308cf24884cee120c378f6. Now, we have `VirtualFileSystem` that implements `org.enso.filesystem.FileSystem<java.nio.file.Path>`. It is also much easier to use.